src/status_im/contexts/wallet/events.cljs

```diff
@@ -386,3 +387,29 @@
             {:title   title
              :subject title
              :message content})]]}))
+
+(rf/reg-event-fx :wallet/secret-phrase
+ (fn [{:keys [db]} [{:keys [secret-phrase random-phrase]}]]
+   {:db (-> db
+            (assoc-in [:wallet :ui :create-account :secret-phrase] secret-phrase)
+            (assoc-in [:wallet :ui :create-account :random-phrase] random-phrase))
+    :fx [[:dispatch-later [{:ms 20 :dispatch [:navigate-to :wallet-check-your-backup]}]]]}))
+
+(rf/reg-event-fx :wallet/new-keypair-created
+ (fn [{:keys [db]} [{:keys [new-keypair]}]]
+   {:db (-> db
+            (assoc-in [:wallet :ui :create-account :new-keypair] new-keypair))
+    :fx [[:dispatch [:navigate-back-to :wallet-create-account]]]}))
+
+(rf/reg-event-fx :wallet/new-keypair-continue
+ (fn [{:keys [db]} [{:keys [keypair-name]}]]
+   (let [secret-phrase (get-in db [:wallet :ui :create-account :secret-phrase])]
+     (native-module/create-account-from-mnemonic
+      {:MnemonicPhrase (string/join " " secret-phrase)}
```

Comment:
I think this should be inside an effect (`reg-fx`) and not an event, since it seems to have side-effects by generating an account on the status-go side.

Reply:
Thanks @clauxx, fixed! 👍